src/app/templates/PaymentProviderInput/PaymentProvidersMenu/index.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { FC, useEffect } from 'react';
+
+import DropdownWrapper from 'app/atoms/DropdownWrapper';
+import Spinner from 'app/atoms/Spinner/Spinner';
+import { useAppEnvStyle } from 'app/hooks/use-app-env-style.hook';
+import { AnalyticsEventCategory, TestIDProperty, useAnalytics } from 'lib/analytics';
+import { PaymentProviderInterface } from 'lib/buy-with-credit-card/topup.interface';
+import { T } from 'lib/i18n';
+
+import { PaymentProviderOption } from './PaymentProviderOption';
+
+interface Props extends TestIDProperty {
+  value?: PaymentProviderInterface;
+  options: PaymentProviderInterface[];
+  isLoading?: boolean;
+  opened: boolean;
+  setOpened: (newValue: boolean) => void;
+  onChange?: (newValue: PaymentProviderInterface) => void;
+}
+
+export const PaymentProvidersMenu: FC<Props> = ({
+  value,
+  options,
+  isLoading = false,
+  opened,
+  testID,
+  setOpened,
+  onChange
+}) => {
+  const { dropdownWidth } = useAppEnvStyle();
+
+  const { trackEvent } = useAnalytics();
+
+  useEffect(() => {
+    if (testID && opened) trackEvent(testID, AnalyticsEventCategory.DropdownOpened);
+  }, [opened, trackEvent]);
+
+  const handleOptionClick = onChange
+    ? (newValue: PaymentProviderInterface) => {
+        if (value?.id !== newValue.id) {
+          onChange(newValue);
+        }
+        setOpened(false);
+      }
+    : undefined;
+
+  return (
+    <DropdownWrapper
+      opened={opened}
+      className="origin-top overflow-x-hidden overflow-y-auto p-2"
+      style={{
```

Comment:
Can we avoid inline styles here?

Reply:
Actually, there is no way to avoid them completely because some styles are defined as inline here: https://github.com/madfish-solutions/templewallet-extension/blob/db08f317df7169a4faa3c306905363810849d0fe/src/app/atoms/DropdownWrapper.tsx#L41